source/fundamentals/crud/compound-operations.txt

```diff
@@ -0,0 +1,309 @@
+to the ``FindOneAndUpdate()`` function:
+
+- A ``Context`` type 
+- A ``non-nil`` query filter specifying which documents to match
+- An update document specifying at least one update operator with fields and values that describe the change 
+
+The function optionally takes a ``FineOneAndUpdateOptions`` type as a
+fourth parameter, which represents options you can use to configure the
+compound operation. If you don't specify a ``FineOneAndUpdateOptions``,
+the driver uses the default values for each option.
+
+The ``FineOneAndUpdateOptions`` type allows you to configure options
+with the following functions:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Function
+     - Description
+
+   * - ``SetArrayFilters()``
+     - | The array elements the update applies to.
+       | Default: ``nil``
+
+   * - ``SetBypassDocumentValidation()``
+     - | Whether to allow the write operation to opt-out of :manual:`document level validation </core/schema-validation>`.
+       | Default: ``false``
+
+   * - ``SetCollation()``
+     - | The type of language collation to use when sorting results.
+       | Default: ``nil``
+       
+   * - ``SetMaxTime()``
+     - | The maximum amount of time that the query can run on the server.
+       | Default: ``nil``
+
+   * - ``SetProjection()``
+     - | The fields to include in the document returned.
+       | Default: ``nil``
+
+   * - ``SetReturnDocument()``
+     - | Whether to return the original or updated document in the ``SingleResult``. 
+       | Default: ``options.Before``
+
+   * - ``SetSort()``
+     - | The sort fields and directions to order the documents matched. 
+       | Default: ``nil``
+
+   * - ``SetUpsert()``
+     - | Whether to insert a new document if the query filter doesn't match any documents. 
+       | Default: ``false``
+
+   * - ``SetHint()``
+     - | The index to use to scan for documents.
+       | Default: ``nil``
+
+By default, the ``FindOneAndUpdate()`` function returns a ``SingleResult`` 
+containing original document. If you configure the ``SetReturnDocument()``
+function to ``options.After``, the function returns the updated document.
+
+Example
+~~~~~~~
+
+The following example performs the following actions in order with the
+``FindOneAndUpdate()`` function:
+
+- Matches a document where the ``type`` is ``"Oolong"``
+- Updates the matched document's ``rating`` to ``9``
+- Returns the updated document
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/compoundOperations.go
+   :language: go
+   :dedent:
+   :start-after: begin FindOneAndUpdate
+   :end-before: end FindOneAndUpdate
+
+After running this example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   [{_id ObjectID("...")} {type Oolong} {rating 9}]
+
+.. _go-find-and-replace:
+
+Find and Replace
+----------------
+
+To find and update one document, pass the following parameters in order
+to the ``FindOneAndReplace()`` function:
```

Comment:
Same as above, more description about what `FindOneAndReplace` does would be helpful here. Specifically, how does it differ from `ReplaceOne` and does it return the found document before or after the replace?